### PR TITLE
8282511: Use fixed certificate validation date in SSLExampleCert template

### PR DESCRIPTION
**Pull request by wkia**
Test files changed only.

The original changes use JDK17 strings in SSLExampleCert.java, so it was adapted for JDK8 with the additional commit as it was done for JDK11 backport.

Additionaly the test uses 'hosts' file substitution that is not available in JDK8. So, it was replaced with a custom NameService implementation.

There were 2 changes with test files paths:

- Original `test/jdk/javax/net/ssl/ServerName/EndingDotHostname.java` path changed to `jdk/test/javax/net/ssl/ServerName/EndingDotHostname.java`.
- Original `test/jdk/javax/net/ssl/templates/SSLExampleCert.java` path changed to `jdk/test/javax/net/ssl/templates/SSLExampleCert.java`.

Please note that this PR is a part of dependant PR chain
Prev PRs:
- #205 
- #206

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed (2 reviews required, with at least 2 [Reviewers](https://openjdk.org/bylaws#reviewer))

### Issues
 * [JDK-8282511](https://bugs.openjdk.org/browse/JDK-8282511): Use fixed certificate validation date in SSLExampleCert template
 * [JDK-8299445](https://bugs.openjdk.org/browse/JDK-8299445): EndingDotHostname.java fails because of compilation errors

### Reviewers
 * [Yuri Nesterenko](https://openjdk.org/census#yan) (@yan-too - **Reviewer**) ⚠️ Review applies to [d71de914](https://git.openjdk.org/jdk8u-dev/pull/207/files/d71de9145a982f04add034e289db5d99272a1f9c)
 * [Andrew John Hughes](https://openjdk.org/census#andrew) (@gnu-andrew - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk8u-dev pull/207/head:pull/207` \
`$ git checkout pull/207`

Update a local copy of the PR: \
`$ git checkout pull/207` \
`$ git pull https://git.openjdk.org/jdk8u-dev pull/207/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 207`

View PR using the GUI difftool: \
`$ git pr show -t 207`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk8u-dev/pull/207.diff">https://git.openjdk.org/jdk8u-dev/pull/207.diff</a>

</details>

### CODE REVIEW

**Comment by wkia** [comment]
The original changes use JDK17 strings in SSLExampleCert.java, so it was adapted for JDK8 with the additional commit as it was done for JDK11 backport.

Additionaly the test uses 'hosts' file substitution that is not available in JDK8. So, it was replaced with a custom NameService implementation.

Force-pushed to keep the commit history clean to make reviewing easier.

**Comment by wkia** [comment]
@gnu-andrew Could you also review this, please?

**Review by yan-too — APPROVED** [review]

**Comment by wkia** [comment]
@gnu-andrew Could you re-review, please?

**Comment by gnu-andrew** [comment]
> The original changes use JDK17 strings in SSLExampleCert.java, so it was adapted for JDK8 with the additional commit as it was done for JDK11 backport.
> 

As far as I can see, the [11u backport](https://github.com/openjdk/jdk11u-dev/commit/3e65a7c5d35b82e14b549ee8ea4bf3a49cfec2e3#diff-50dba36cc764382a6c70482237ced2794f3c7edf695b596a5b64070d4090fb74) still had text blocks. Do you mean you've also included the changes in [JDK-8299445](https://bugs.openjdk.org/browse/JDK-8299445) here? If so, we should reference it too.

> Additionaly the test uses 'hosts' file substitution that is not available in JDK8. So, it was replaced with a custom NameService implementation.
> 
> Force-pushed to keep the commit history clean to make reviewing easier.

Thanks for removing the superfluous module line.

Is there a reason the custom `NameService` creates an `InetAddress` using an internal API rather than just returning `InetAddress.getLoopbackAddress()` which would be closer to the 11u code?

Has the test been run successfully on 8u? The follow-up fix on 11u suggests this wasn't done before committing the fix to 11u.

**Comment by wkia** [comment]
> As far as I can see, the [11u backport](https://github.com/openjdk/jdk11u-dev/commit/3e65a7c5d35b82e14b549ee8ea4bf3a49cfec2e3#diff-50dba36cc764382a6c70482237ced2794f3c7edf695b596a5b64070d4090fb74) still had text blocks. Do you mean you've also included the changes in [JDK-8299445](https://bugs.openjdk.org/browse/JDK-8299445) here? If so, we should reference it too.

Thanks, I added the reference.

> Is there a reason the custom `NameService` creates an `InetAddress` using an internal API rather than just returning `InetAddress.getLoopbackAddress()` which would be closer to the 11u code?

Sounds reasonable, thanks. I've updated the code.

> Has the test been run successfully on 8u? The follow-up fix on 11u suggests this wasn't done before committing the fix to 11u.

Yes, it passes successfully on my Win11/Cygwin.

**Review by gnu-andrew — APPROVED** [review]
Thanks, that looks a lot simpler.

Yes, `jdk8u-fix-request` can be added to the bug now all reviews are complete. Once you've done that, I'll approve and you can run the `integrate` command.

Thanks for persisting with this one.

File: jdk/test/javax/net/ssl/ServerName/EndingDotHostname.java [hunk 1]
```diff
@@ -26,7 +26,8 @@
  * @bug 8065422
  * @summary Trailing dot in hostname causes TLS handshake to fail
  * @library /javax/net/ssl/templates
- * @run main/othervm -Djdk.net.hosts.file=hostsForExample EndingDotHostname
+ * @run main/othervm --add-opens java.base/sun.security.ssl=ALL-UNNAMED
```
[the remark this answers is not in the document]
Comment (wkia) [reply jdk/test/javax/net/ssl/ServerName/EndingDotHostname.java]: The finalchnages don't contain `--add-opens`. The test passes on 8u.

**Pull request closed without merging**